src/pages/filter/ui/modal-edit/components/modal-inheritance-mode/modal-inheritance-mode.tsx

```diff
@@ -0,0 +1,114 @@
+import { ReactElement, useEffect, useRef, useState } from 'react'
+import { observer } from 'mobx-react-lite'
+
+import { ActionType } from '@declarations'
+import dtreeStore from '@store/dtree'
+import { InheritanceModeContent } from '@pages/filter/ui/modal-edit/components/modal-inheritance-mode/components/inheritance-mode-content'
+import { HeaderModal } from '@pages/filter/ui/query-builder/ui/header-modal'
+import { ModalBase } from '@pages/filter/ui/query-builder/ui/modal-base'
+import { SelectModalButtons } from '@pages/filter/ui/query-builder/ui/select-modal-buttons'
+import modalEditStore from '../../modal-edit.store'
+import { EditModalButtons } from '../edit-modal-buttons'
+import modalInheritanceModeStore from './modal-inheritance-mode.store'
+
+export const ModalInheritanceMode = observer((): ReactElement => {
+  const ref = useRef(null)
+
+  const {
+    currentStepIndex,
+    currentGroupIndex,
+    currentGroup,
+    groupName,
+    selectedGroupsAmount,
+    problemGroups,
+  } = modalEditStore
+
+  const [selectedProblemGroups, setSelectedProblemGroups] = useState<string[]>(
+    currentGroup
+      ? (Object.values(currentGroup[currentGroup.length - 1])[0] as string[])
+      : [],
+  )
+
+  useEffect(() => {
+    if (currentGroup) {
+      dtreeStore.stepData[currentStepIndex].groups[currentGroupIndex]
+        .find((elem: any) => Array.isArray(elem))
+        .map((item: string) => dtreeStore.addSelectedFilter(item))
+    }
+
+    return () => {
+      dtreeStore.resetSelectedFilters()
+    }
+  }, [currentGroup, currentGroupIndex, currentStepIndex])
+
+  useEffect(() => {
+    const params = `{"problem_group":["${selectedProblemGroups
+      .toString()
+      .split(',')
+      .join('","')}"]}`
+
+    const initAsync = async () => {
+      await dtreeStore.fetchStatFuncAsync(groupName, params)
+    }
+
+    initAsync()
```

Comment:
Avoid bringing logic into views, better to create an extra method in the store, like you did on the 60th line